js/packages/web/src/actions/nft.tsx

```diff
@@ -121,33 +124,71 @@ export const mintNFT = async (
     },
   };
 
-  const realFiles: File[] = [
-    ...files,
-    new File([JSON.stringify(metadataContent)], RESERVED_METADATA),
-  ];
+  const realFiles: File[] = [...files]
+  const fileDataForm = new FormData();
 
-  const { instructions: pushInstructions, signers: pushSigners } =
-    await prepPayForFilesTxn(wallet, realFiles, metadata);
+  realFiles.map((f) => {
+    fileDataForm.append(
+      `file[${f.name}]`,
+      f,
+      f.name
+    )
+  });
+
+  const uploadResponse = await fetch(
+    nftStorageHolaplexEndpoint,
+    {
+      method: "POST",
+      body: fileDataForm,
+    })
+  const uploadedFilePins = await uploadResponse.json()
+  // add files to properties
+  // first image is added as image
 
   progressCallback(1)
+  let imageSet = false;
+  metadataContent.properties.files = []
+  uploadedFilePins.files.forEach((file) => {
+    if (!coverFile && !imageSet && /image/.test(file.type)) {
+      metadataContent.image = file.uri
+      imageSet = true;
+    }
+    metadataContent.properties.files.push({
+      uri: file.uri,
+      type: file.type
+    })
+  })
```

Comment:
This is a good spot for a reduce.

const metadataContent.properties.files = uploadFilePins.files.reduce((memo, file) => { ... }, [])